packages/virtual/src/index.tsx

```diff
@@ -1,71 +1,109 @@
 import { For, createSignal } from "solid-js";
-import type { Accessor, JSX } from "solid-js";
+import type { Accessor, JSX, Signal } from "solid-js";
 
 /**
- * A basic virtualized list (see https://www.patterns.dev/vanilla/virtual-lists/) component for improving performance when rendering very large lists
+ * A headless virtualized list (see https://www.patterns.dev/vanilla/virtual-lists/) utility for constructing your own virtualized list components with maximum flexibility.
+ *
+ * @param rootElement accessor for the element to use as the root of the virtualized list
+ * @param items the list of items
+ * @param rootHeight the height of the root element of the virtualizedList
+ * @param rowHeight the height of individual rows in the virtualizedList
+ * @param overscanCount the number of elements to render both before and after the visible section of the list, so passing 5 will render 5 items before the list, and 5 items after. Defaults to 1, cannot be set to zero. This is necessary to hide the blank space around list items when scrolling
+ * @returns an object whose properties are used by the list's components
+ */
+export function createVirtualList<T extends readonly any[]>({
+  rootElement,
+  items,
+  rootHeight,
+  rowHeight,
+  overscanCount,
+}: {
+  rootElement: Accessor<Element>;
```

Comment:
Instead of having both `rootElement: Accessor<Element>` and `onScroll: VoidFunction`, I think we should stick to one of those:
- `rootElement: Accessor<Element | undefined | null | false>` and we add an event listener in an effect ourselves, no need for handling scroll by the user.
- `offset: Accessor<number>` and the user instead of creating a signal for the element, created one for the scrollTop value and manages that - it's the same amount of code for the user anyway
- `onScroll: (el: Element) => void` and we read the scrollTop from the element and the user adds the listener
- `onScroll: (offset: number) => void` and the user reads the scrollTop from the element and adds the listener

Reply:
I ended up with not passing in the root element and using `onScroll(event: ScrollEvent) => void` and having the utility get the scrollTop from `event.target`. The typescript for it is relatively straightforwards and it's the easiest for the end user.